### PR TITLE
[rag eval] Add navigation logic and get calls feedback working

### PR DESCRIPTION
**Pull request by chejennifer**
https://github.com/datacommonsorg/website/assets/69875368/29977c28-21ad-4ebf-8bdd-7f6f8300b3ca

TODO: add loading spinner & maybe look into why the feedback section is slow to update

### CODE REVIEW

File: static/js/apps/eval_retrieval_generation/feedback_navigation.tsx [hunk 1]
```diff
@@ -25,10 +25,98 @@ import { AppContext, SessionContext } from "./context";
...
+): boolean {
+  if (feedbackStage === FeedbackStage.CALLS) {
+    return sessionCallId === NEW_QUERY_CALL_ID;
+  }
+  // feedback stages besides CALLS only have one page
+  return true;
+}
+
+// Whether or not to include a stage in the list of stages to show
+function includeStage(feedbackStage: FeedbackStage, numCalls: number) {
```
Comment (shifucun) [inline static/js/apps/eval_retrieval_generation/feedback_navigation.tsx:111]: nit: show bool return type

Comment (shifucun) [reply static/js/apps/eval_retrieval_generation/feedback_navigation.tsx:111]: naming nit: shouldIncludeStage()

File: static/js/apps/eval_retrieval_generation/feedback_navigation.tsx [hunk 2]
```diff
@@ -25,10 +25,98 @@ import { AppContext, SessionContext } from "./context";
...
+  if (feedbackStage === FeedbackStage.CALLS) {
+    return numCalls > 0;
+  }
+  return true;
+}
+
 interface FeedbackNavigationPropType {
   checkAndSubmit: () => Promise<boolean>;
-  // This overrides the logic for whether or not to show the "next" button
-  showNextOverride?: boolean;
```
Comment (shifucun) [inline static/js/apps/eval_retrieval_generation/feedback_navigation.tsx:31]: get rid of override is nice!

File: static/js/apps/eval_retrieval_generation/feedback_navigation.tsx [hunk 3]
```diff
@@ -110,104 +198,102 @@ export function FeedbackNavigation(
...
+      case ButtonType.PREV_EVAL_STAGE:
+        return prevEvalStage;
+      case ButtonType.NEXT_QUERY:
+        return nextQuery;
+      case ButtonType.NEXT:
+        return next;
+      case ButtonType.NEXT_EVAL_STAGE:
+        return nextEvalStage;
+      case ButtonType.FINISH:
+        return props.checkAndSubmit;
```
Comment (shifucun) [inline static/js/apps/eval_retrieval_generation/feedback_navigation.tsx:252]: default throw error?

File: static/js/apps/eval_retrieval_generation/query_section.tsx [hunk 4]
```diff
@@ -66,9 +66,15 @@ export function QuerySection(): JSX.Element {
...
         const row = rows[0];
-        const rowQ = `**${row.get(DC_QUESTION_COL)}**`;
-        const rowA = `${row.get(DC_RESPONSE_COL)} \xb7 Table ${tableIds[i]}`;
-        answers.push(`${rowQ}\n\n${rowA}`);
+        const rowQ = `<span class="dc-question">**${row.get(
+          DC_QUESTION_COL
+        )}**</span>`;
+        const rowA = `<span class="dc-stat">${row.get(
+          DC_RESPONSE_COL
+        )} \xb7 Table ${tableIds[i]}</span>`;
```
Comment (shifucun) [inline static/js/apps/eval_retrieval_generation/query_section.tsx:74]: maybe have separate vars for row.get(xx); current string formatting is a bit hard to follow.

**Review by shifucun — COMMENTED** [review]
nice refactor and re-organize!

File: static/js/apps/eval_retrieval_generation/feedback_navigation.tsx (hunk 3 above)
Comment (chejennifer) [reply static/js/apps/eval_retrieval_generation/feedback_navigation.tsx:252]: I don't think this would be an error, but instead added a default case with noop

File: static/js/apps/eval_retrieval_generation/feedback_navigation.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/apps/eval_retrieval_generation/feedback_navigation.tsx]: done

File: static/js/apps/eval_retrieval_generation/query_section.tsx (hunk 4 above)
Comment (chejennifer) [reply static/js/apps/eval_retrieval_generation/query_section.tsx]: done

**Review by shifucun — APPROVED** [review]
quick and nice!

**Comment by chejennifer** [comment]
thanks for the review!
